frontend/awx/administration/settings/AwxSettingsForm.tsx

```diff
@@ -160,12 +162,39 @@ export function AwxSettingsForm(props: {
       return acc;
     }, {});
 
+  function getCategorySlugs(config: Record<string, AwxSettingsOptionsAction>): string[] {
+    const slugs = new Set<string>();
+
+    Object.values(config).forEach((item) => {
+      const slug = item?.category_slug;
+      if (slug) {
+        slugs.add(slug);
+      }
+    });
+
+    return Array.from(slugs);
+  }
+
```

Comment:
`getCategorySlugs ` provides a list of the unique categories for the settings in the form. 

**Example**: 
For the GitHub Enterprise OAuth2 form located at "/access/authentication/github-enterprise-oauth2/edit", it will return `[github-enterprise, github-enterprise-org, github-enterprise-team]`. The "Revert All" modal will use this list to go through each category and send a DELETE request for each one.

Reply:
Good explanation